.github/workflows/release.yaml

```diff
@@ -0,0 +1,142 @@
+          mv bin/azure-cloud-node-manager bin/azure-cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}
+      - name: Build binary for windows
+        if: matrix.os == 'windows'
+        run: |
+          rm -rf ./bin
+          ARCH=${{ matrix.arch }} make bin/azure-cloud-node-manager.exe
+          mv bin/azure-cloud-node-manager-${{ matrix.arch }}.exe bin/azure-cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}.exe
+      - name: Upload artifact for linux
+        if: matrix.os == 'linux'
+        uses: actions/upload-artifact@v2
+        with:
+          name: cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}
+          path: bin/azure-cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}
+      - name: Upload artifact for windows
+        if: matrix.os == 'windows'
+        uses: actions/upload-artifact@v2
+        with:
+          name: cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}
+          path: bin/azure-cloud-node-manager-${{ matrix.os }}-${{ matrix.arch }}.exe
+          if-no-files-found: error
+
+  publish:
+    runs-on: ubuntu-latest
+    needs:
+      - build-cloud-controller-manager
+      - build-cloud-node-manager
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
+      - name: Generate release note
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          VERSION="${GITHUB_REF_NAME#[vV]}"
+          VERSION_MAJOR="${VERSION%%\.*}"
+          VERSION_MINOR="${VERSION#*.}"
+          VERSION_MINOR="${VERSION_MINOR%.*}"
+          VERSION_PATCH="${VERSION##*.}"
+          
+          BRANCH=release-${VERSION_MAJOR}.${VERSION_MINOR}
+          if [[ "${VERSION_PATCH}" = "0" ]]; then
+            START_SHA=$(git rev-parse v${VERSION_MAJOR}.$((VERSION_MINOR-1)).0)
+          else
+            START_SHA=$(git rev-parse v${VERSION_MAJOR}.${VERSION_MINOR}.$((VERSION_PATCH-1)))
+          fi
+          END_SHA=$(git rev-parse ${GITHUB_REF_NAME})
+          
+          echo ${BRANCH}
+          echo ${START_SHA}
+          echo ${END_SHA}
+          
+          GO111MODULE=on go install k8s.io/release/cmd/release-notes@latest
+          release-notes --repo=cloud-provider-azure \
+          	--org=kubernetes-sigs \
+          	--branch=${BRANCH} \
+          	--start-sha=${START_SHA} \
+          	--end-sha=${END_SHA} \
+          	--output=release-note.md
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: ./artifacts
+      - name: Publish release
+        uses: softprops/action-gh-release@v1
+        with:
+          body_path: release-note.md
+          files: |
+            ./artifacts/cloud-node-manager-*-*/*
+            ./artifacts/cloud-controller-manager-*-*/*
```

Comment:
I'm wondering if building docker images will be a more feasible option.

Reply:
please take a look at https://github.com/kubernetes-sigs/cloud-provider-azure/issues/879. 
Docker images are available in [MCR](https://mcr.microsoft.com/v2/oss/kubernetes/azure-cloud-node-manager/tags/list), and this PR provides binary as another option.

Reply:
yea, we only need binaries here. Images would be pushed to MCR